mermithid/processors/misc/MultiGasComplexLineShape.py

```diff
@@ -0,0 +1,3748 @@
+        if emitted_peak == 'lorentzian':
+            current_working_spectrum = self.std_lorenztian_17keV()
+        elif emitted_peak == 'shake':
+            current_working_spectrum = self.shakeSpectrumClassInstance.shake_spectrum()
+        elif emitted_peak == 'dirac':
+            current_working_spectrum = self.std_dirac()
+
+        if self.use_combined_four_trap_inst_reso:
+            current_working_spectrum = self.convolve_ins_resolution_combining_four_trap(current_working_spectrum, self.trap_weights)
+        else:
+            current_working_spectrum = self.convolve_ins_resolution(current_working_spectrum)
+
+        zeroth_order_peak = current_working_spectrum
+        current_full_spectrum += current_working_spectrum
+        N = len(self.gases)
+        for M in range(1, self.max_scatters + 1):
+            gas_scatter_combinations = np.array([np.array(i) for i in product(range(M+1), repeat=N) if sum(i)==M])
+            for combination in gas_scatter_combinations:
+                entry_str = ''
+                for component, gas_type in zip(combination, self.gases):
+                    entry_str += gas_type
+                    entry_str += str(component).zfill(2)
+                current_working_spectrum = scatter_spectra.item()[entry_str]
+                current_working_spectrum = self.normalize(signal.convolve(zeroth_order_peak, current_working_spectrum, mode='same'))
+                coefficient = factorial(sum(combination))
+                for component, i in zip(combination, range(len(self.gases))):
+                    coefficient = coefficient/factorial(component)*p[i]**component
+                current_full_spectrum += coefficient*current_working_spectrum*prob_parameter**M
+        return current_full_spectrum
+
+    def spectrum_func_ftc_2(self, bins_Hz, *p0):
+        B_field = p0[0]
+        amplitude = p0[1]
+        prob_parameter = p0[2]
+        N = len(self.gases)
+        scatter_proportion = p0[3:2+N]
+
+        x_eV = ConversionFunctions.Energy(bins_Hz, B_field)
+        en_loss_array = self.std_eV_array()
+        en_loss_array_min = en_loss_array[0]
+        en_loss_array_max = en_loss_array[len(en_loss_array)-1]
+        en_array_rev = ComplexLineShapeUtilities.flip_array(-1*en_loss_array)
+        f = np.zeros(len(x_eV))
+        f_intermediate = np.zeros(len(x_eV))
+
+        x_eV_minus_line = Constants.kr_k_line_e() - x_eV
+        zero_idx = np.r_[np.where(x_eV_minus_line< en_loss_array_min)[0],np.where(x_eV_minus_line>en_loss_array_max)[0]]
+        nonzero_idx = [i for i in range(len(x_eV)) if i not in zero_idx]
+
+        full_spectrum = self.make_spectrum_ftc_2(prob_parameter, scatter_proportion)
+        f_intermediate[nonzero_idx] = np.interp(x_eV_minus_line[nonzero_idx],en_array_rev,full_spectrum)
+        f[nonzero_idx] += amplitude*f_intermediate[nonzero_idx]/np.sum(f_intermediate[nonzero_idx])
+        return f
+
+    def fit_data_ftc_2(self, freq_bins, data_hist_freq):
```

Comment:
what's the difference between fit_data_fct and fit_data_ftc_2

Reply:
The difference between the fit_data_fct() and fit_data_ftc2() is that in fit_data_ftc2() the modified exponential function is not included in fit_data_ftc2(). A comment is added at around line 1235 to explain this.